{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b5fcbe25",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "44fff119",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>\n",
      "distutils: /opt/conda/include/python3.8/UNKNOWN\n",
      "sysconfig: /opt/conda/include/python3.8\u001b[0m\n",
      "\u001b[33mWARNING: Additional context:\n",
      "user = False\n",
      "home = None\n",
      "root = None\n",
      "prefix = None\u001b[0m\n",
      "Requirement already satisfied: biopython in /opt/conda/lib/python3.8/site-packages (1.79)\n",
      "Requirement already satisfied: GEOparse in /opt/conda/lib/python3.8/site-packages (2.0.3)\n",
      "Requirement already satisfied: biomart in /opt/conda/lib/python3.8/site-packages (0.9.2)\n",
      "Collecting nltk\n",
      "  Downloading nltk-3.7-py3-none-any.whl (1.5 MB)\n",
      "\u001b[K     |████████████████████████████████| 1.5 MB 264 kB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: numpy in /opt/conda/lib/python3.8/site-packages (from biopython) (1.19.5)\n",
      "Requirement already satisfied: pandas>=0.17 in /opt/conda/lib/python3.8/site-packages (from GEOparse) (1.2.4)\n",
      "Requirement already satisfied: requests>=2.21.0 in /opt/conda/lib/python3.8/site-packages (from GEOparse) (2.25.1)\n",
      "Requirement already satisfied: tqdm>=4.31.1 in /opt/conda/lib/python3.8/site-packages (from GEOparse) (4.60.0)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /opt/conda/lib/python3.8/site-packages (from pandas>=0.17->GEOparse) (2.8.1)\n",
      "Requirement already satisfied: pytz>=2017.3 in /opt/conda/lib/python3.8/site-packages (from pandas>=0.17->GEOparse) (2021.1)\n",
      "Requirement already satisfied: six>=1.5 in /opt/conda/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas>=0.17->GEOparse) (1.15.0)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in /opt/conda/lib/python3.8/site-packages (from requests>=2.21.0->GEOparse) (4.0.0)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /opt/conda/lib/python3.8/site-packages (from requests>=2.21.0->GEOparse) (2.10)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.8/site-packages (from requests>=2.21.0->GEOparse) (1.26.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.8/site-packages (from requests>=2.21.0->GEOparse) (2020.12.5)\n",
      "Requirement already satisfied: joblib in /opt/conda/lib/python3.8/site-packages (from nltk) (1.0.1)\n",
      "Collecting regex>=2021.8.3\n",
      "  Downloading regex-2022.10.31-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (772 kB)\n",
      "\u001b[K     |████████████████████████████████| 772 kB 2.5 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: click in /opt/conda/lib/python3.8/site-packages (from nltk) (7.1.2)\n",
      "Installing collected packages: regex, nltk\n",
      "\u001b[33mWARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>\n",
      "distutils: /opt/conda/include/python3.8/UNKNOWN\n",
      "sysconfig: /opt/conda/include/python3.8\u001b[0m\n",
      "\u001b[33mWARNING: Additional context:\n",
      "user = False\n",
      "home = None\n",
      "root = None\n",
      "prefix = None\u001b[0m\n",
      "Successfully installed nltk-3.7 regex-2022.10.31\n"
     ]
    }
   ],
   "source": [
    "!{sys.executable} -m pip install biopython GEOparse biomart nltk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "ab593b77",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "import GEOparse\n",
    "import pandas as pd\n",
    "import re\n",
    "import biomart\n",
    "import nltk\n",
    "from nltk.tokenize import word_tokenize  \n",
    "import json\n",
    "from os import path\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "3e9c6cd7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to /home/jovyan/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nltk.download('punkt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f2f60f0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = \"emerson@snthesis.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "bb532f96",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_ensembl_mappings():                                                    \n",
    "    server = biomart.BiomartServer('http://www.ensembl.org/biomart', verbose=True)         \n",
    "    mart = server.datasets['hsapiens_gene_ensembl']  \n",
    "    \n",
    "    attributes = [\n",
    "    \"affy_hg_u133_plus_2\",\n",
    "    \"ensembl_gene_id\",\n",
    "    \"entrezgene_id\",\n",
    "    \"external_gene_name\"]\n",
    "                                                                                \n",
    "    # Get the mapping between the attributes                                    \n",
    "    response = mart.search({'attributes': attributes})                          \n",
    "    data = response.raw.data.decode('ascii')                                    \n",
    "                                                                                \n",
    "    ensembl_to_genesymbol = {}                                                  \n",
    "    # Store the data in a dict                                                  \n",
    "    for line in data.splitlines():                                              \n",
    "        line = line.split('\\t')                                                 \n",
    "        # The entries are in the same order as in the `attributes` variable\n",
    "        probe_id = line[0]                                                 \n",
    "        ensembl_gene_id = line[1]                                                   \n",
    "        entrez_id = line[2]                                                  \n",
    "        external_gene_name = line[3]                                               \n",
    "                                                                                \n",
    "        # Some of these keys may be an empty string. If you want, you can \n",
    "        # avoid having a '' key in your dict by ensuring the \n",
    "        # transcript/gene/peptide ids have a nonzero length before\n",
    "        # adding them to the dict\n",
    "        ensembl_to_genesymbol[probe_id] = entrez_id                      \n",
    "        ensembl_to_genesymbol[ensembl_gene_id] = entrez_id                       \n",
    "        ensembl_to_genesymbol[external_gene_name] = entrez_id                \n",
    "                                                                                \n",
    "    return ensembl_to_genesymbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "a13f6be6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartServer:'http://www.ensembl.org/biomart/martservice'] is alive.\n",
      "[BiomartServer:'http://www.ensembl.org/biomart/martservice'] Fetching datasets\n",
      "[BiomartServer:'http://www.ensembl.org/biomart/martservice'] Fetching databases\n",
      "[BiomartDatabase:'Ensembl Genes 108'] Fetching datasets\n",
      "[BiomartDatabase:'Mouse strains 108'] Fetching datasets\n",
      "[BiomartDatabase:'Sequence'] Fetching datasets\n",
      "[BiomartDatabase:'Ontology'] Fetching datasets\n",
      "[BiomartDatabase:'Genomic features 108'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Variation 108'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Regulation 108'] Fetching datasets\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Searching using following params:\n",
      "{'attributes': ['affy_hg_u133_plus_2',\n",
      "                'ensembl_gene_id',\n",
      "                'entrezgene_id',\n",
      "                'external_gene_name']}\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "[BiomartDataset] search query:\n",
      "b'<Query virtualSchemaName=\"default\" formatter=\"TSV\" header=\"0\" uniqueRows=\"1\" datasetConfigVersion=\"0.6\" count=\"\"><Dataset name=\"hsapiens_gene_ensembl\" interface=\"default\"><Attribute name=\"affy_hg_u133_plus_2\" /><Attribute name=\"ensembl_gene_id\" /><Attribute name=\"entrezgene_id\" /><Attribute name=\"external_gene_name\" /></Dataset></Query>'\n"
     ]
    }
   ],
   "source": [
    "if not path.exists('./probe_mappings_affy_hg_u133_plus_2.json'):\n",
    "    probe_mappings = get_ensembl_mappings()\n",
    "    with open('./probe_mappings_affy_hg_u133_plus_2.json', 'w') as file:\n",
    "        file.write(json.dumps(mappings))\n",
    "else:\n",
    "    with open('./probe_mappings_affy_hg_u133_plus_2.json') as file:\n",
    "        probe_mappings = json.load(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "9acf6503",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "09-Nov-2022 20:00:36 DEBUG utils - Directory ./geo_data already exists. Skipping.\n",
      "09-Nov-2022 20:00:36 INFO GEOparse - File already exist: using local version.\n",
      "09-Nov-2022 20:00:36 INFO GEOparse - Parsing ./geo_data/GSE72970_family.soft.gz: \n",
      "09-Nov-2022 20:00:36 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "09-Nov-2022 20:00:36 DEBUG GEOparse - SERIES: GSE72970\n",
      "09-Nov-2022 20:00:36 DEBUG GEOparse - PLATFORM: GPL570\n",
      "/opt/conda/lib/python3.8/site-packages/GEOparse/GEOparse.py:113: DtypeWarning: Columns (2) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  return parse_GSE(filepath, open_kwargs=open_kwargs)\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875897\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875898\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875899\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875900\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875901\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875902\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875903\n",
      "09-Nov-2022 20:00:37 DEBUG GEOparse - SAMPLE: GSM1875904\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875905\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875906\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875907\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875908\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875909\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875910\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875911\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875912\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875913\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875914\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875915\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875916\n",
      "09-Nov-2022 20:00:38 DEBUG GEOparse - SAMPLE: GSM1875917\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875918\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875919\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875920\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875921\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875922\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875923\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875924\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875925\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875926\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875927\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875928\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875929\n",
      "09-Nov-2022 20:00:39 DEBUG GEOparse - SAMPLE: GSM1875930\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875931\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875932\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875933\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875934\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875935\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875936\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875937\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875938\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875939\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875940\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875941\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875942\n",
      "09-Nov-2022 20:00:40 DEBUG GEOparse - SAMPLE: GSM1875943\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875944\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875945\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875946\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875947\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875948\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875949\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875950\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875951\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875952\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875953\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875954\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875955\n",
      "09-Nov-2022 20:00:41 DEBUG GEOparse - SAMPLE: GSM1875956\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875957\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875958\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875959\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875960\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875961\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875962\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875963\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875964\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875965\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875966\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875967\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875968\n",
      "09-Nov-2022 20:00:42 DEBUG GEOparse - SAMPLE: GSM1875969\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875970\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875971\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875972\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875973\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875974\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875975\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875976\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875977\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875978\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875979\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875980\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875981\n",
      "09-Nov-2022 20:00:43 DEBUG GEOparse - SAMPLE: GSM1875982\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875983\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875984\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875985\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875986\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875987\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875988\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875989\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875990\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875991\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875992\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875993\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875994\n",
      "09-Nov-2022 20:00:44 DEBUG GEOparse - SAMPLE: GSM1875995\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1875996\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1875997\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1875998\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1875999\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876000\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876001\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876002\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876003\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876004\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876005\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876006\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876007\n",
      "09-Nov-2022 20:00:45 DEBUG GEOparse - SAMPLE: GSM1876008\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876009\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876010\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876011\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876012\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876013\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876014\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876015\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876016\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876017\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876018\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876019\n",
      "09-Nov-2022 20:00:46 DEBUG GEOparse - SAMPLE: GSM1876020\n"
     ]
    }
   ],
   "source": [
    "series = 'GSE72970'\n",
    "gse = GEOparse.get_GEO(geo=series, destdir=\"./geo_data\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddb4ff70",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43cf34ff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4c136eef",
   "metadata": {},
   "outputs": [],
   "source": [
    "short = [(name, gsm) for index, (name, gsm) in enumerate(gse.gsms.items()) if index < 3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5aa3e07b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__metaclass__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_get_columns_as_string',\n",
       " '_get_metadata_as_string',\n",
       " '_get_object_as_soft',\n",
       " '_get_table_as_string',\n",
       " 'annotate',\n",
       " 'annotate_and_average',\n",
       " 'columns',\n",
       " 'download_SRA',\n",
       " 'download_supplementary_files',\n",
       " 'geotype',\n",
       " 'get_accession',\n",
       " 'get_metadata_attribute',\n",
       " 'get_type',\n",
       " 'head',\n",
       " 'metadata',\n",
       " 'name',\n",
       " 'relations',\n",
       " 'show_columns',\n",
       " 'show_metadata',\n",
       " 'show_table',\n",
       " 'table',\n",
       " 'to_soft']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample = short[0][1]\n",
    "dir(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "43407cbd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'title': ['Primary_Tumor 01_010'],\n",
       " 'geo_accession': ['GSM1875897'],\n",
       " 'status': ['Public on Aug 17 2017'],\n",
       " 'submission_date': ['Sep 12 2015'],\n",
       " 'last_update_date': ['Aug 17 2017'],\n",
       " 'type': ['RNA'],\n",
       " 'channel_count': ['1'],\n",
       " 'source_name_ch1': ['Tissue samples taken from colorectal cancer patients'],\n",
       " 'organism_ch1': ['Homo sapiens'],\n",
       " 'taxid_ch1': ['9606'],\n",
       " 'characteristics_ch1': ['center: Tours',\n",
       "  'Sex: Male',\n",
       "  'age: 53',\n",
       "  'who performance status: 0',\n",
       "  'tumor location: Left colon',\n",
       "  'synchronous metastase: Yes',\n",
       "  'pn: pN2',\n",
       "  'pt: pT3',\n",
       "  'regimen: FOLFOX',\n",
       "  'response category: PR',\n",
       "  'response status: R',\n",
       "  'pfs censored: 1',\n",
       "  'pfs: 5.585215569',\n",
       "  'os censored: 1',\n",
       "  'os: 5.585215569'],\n",
       " 'treatment_protocol_ch1': ['Samples were collected before chemotherapy'],\n",
       " 'molecule_ch1': ['total RNA'],\n",
       " 'extract_protocol_ch1': [\"Rneasy kit (Qiagen) for total RNA was performed according to the manufacturer's instructions.\"],\n",
       " 'label_ch1': ['biotin'],\n",
       " 'label_protocol_ch1': ['Biotinylated cRNA were prepared according to the Affymetrix protocol.'],\n",
       " 'hyb_protocol': ['Following fragmentation, cRNA were hybridized on GeneChip Human Genome U133 Plus 2.0 array.'],\n",
       " 'scan_protocol': ['GeneChips were scanned using a laser scanner.'],\n",
       " 'data_processing': ['Processing and normalization was done using the Robust Multi-array Average (RMA)'],\n",
       " 'platform_id': ['GPL570'],\n",
       " 'contact_name': ['Pierre,,Martineau'],\n",
       " 'contact_email': ['pierre.martineau@inserm.fr'],\n",
       " 'contact_phone': ['+33 4 67 61 37 43'],\n",
       " 'contact_department': ['IRCM'],\n",
       " 'contact_institute': ['Inserm'],\n",
       " 'contact_address': ['208 Rue des Apothicaires'],\n",
       " 'contact_city': ['Montpellier Cedex 5'],\n",
       " 'contact_zip/postal_code': ['34298'],\n",
       " 'contact_country': ['France'],\n",
       " 'supplementary_file': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1875nnn/GSM1875897/suppl/GSM1875897_D60726_VPA10_01_010_3mcgP.CEL.gz'],\n",
       " 'series_id': ['GSE72968', 'GSE72970'],\n",
       " 'data_row_count': ['54675']}"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample.metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "236ed791",
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf = sample.table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "a8958f04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "54675"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sdf['ID_REF'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "b983ba87",
   "metadata": {},
   "outputs": [],
   "source": [
    "def probe_to_entrez_id(probe_id, probe_mappings):\n",
    "    mapped = probe_mappings.get(probe_id, probe_id)\n",
    "    if mapped is None or mapped.strip() == '':\n",
    "        return probe_id\n",
    "    return mapped\n",
    "\n",
    "def to_expression_matrix(dataset, probe_mappings):\n",
    "    items = {item_id: sample.table['VALUE'].tolist() for item_id, sample in dataset.gsms.items()}\n",
    "    item = [sample for i, (_, sample) in enumerate(dataset.gsms.items()) if i < 1][0]\n",
    "    labels = [probe_to_entrez_id(probe_id, probe_mappingsprobe_to_entrez_id) for probe_id in item.table['ID_REF']]\n",
    "    len_labels = len(labels)\n",
    "    items['entrez_id'] = labels\n",
    "    return pd.DataFrame.from_dict(items)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cfdeba40",
   "metadata": {},
   "outputs": [],
   "source": [
    "def safe_get(data, path_item, default={}):\n",
    "    if isinstance(data, dict):\n",
    "        return data.get(path_item, default)\n",
    "    else:\n",
    "        length = len(data)\n",
    "        return data[path_item] if path_item < length else default\n",
    "    \n",
    "\n",
    "def path_mapping(path):\n",
    "    def mapper(data):\n",
    "        for path_item in path[:-1]:\n",
    "            data = safe_get(data, path_item)\n",
    "        return safe_get(data, path[-1], None)\n",
    "    return mapper\n",
    "\n",
    "\n",
    "def value(value):\n",
    "    return lambda _data: value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fc1ef1f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "551b180f",
   "metadata": {},
   "outputs": [],
   "source": [
    "units = set([\n",
    "    \"months\"\n",
    "])\n",
    "responses = {\n",
    "    \"r\": \"response\",\n",
    "}\n",
    "response_categories = {\n",
    "    \"pr\": \"partial_response\",\n",
    "    \"cr\": \"complete_response\",\n",
    "    \"sd\": \"stable_disease\",\n",
    "    \"pd\": \"progressive_disease\"\n",
    "}\n",
    "sample_collected = set([\"before_treatment\", \"after_treatment\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "30437c5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "headers = [\"sample_id\", \"sample_id_type\", \"source_accession\", \"source_type\", \"sex\", \"age\", \"normalized_drug\", \"raw_drug\", \"sample_name\", \"data_processing_notes\", \"sample_collected\", \"response\", \"response_category\", \"progression_free_survival\", \"progression_free_survival_unit\", \"overall_survival\", \"overall_survival_unit\", \"platform_id\", 'tmn_t', 'tmn_n', 'tmn_m']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "773542ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('./drug_synonyms.json') as file:\n",
    "    syn_json = json.load(file)\n",
    "    \n",
    "drop_tokens = set([\n",
    "    'study drug'\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "e2587cd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalize_drug_name(drug_name):\n",
    "    drugs = re.split(r\"\\+|,|/\", drug_name)\n",
    "    for dt in drop_tokens:\n",
    "        drugs = [d.replace(dt, '') for d in drugs]\n",
    "    drugs = [re.sub(r\"(\\d)([A-z])\", \"\\\\1 \\\\2\", d.lower().strip()) for d in drugs]\n",
    "    drugs = [re.sub(r\"(\\d)\\s*-\\s*([A-z])\", \"\\\\1 \\\\2\", d) for d in drugs]\n",
    "    tokens = [\" \".join(word_tokenize(d)) for d in drugs if d not in drop_tokens]\n",
    "    tokens = [syn_json['synonyms'].get(t, t) for t in tokens]\n",
    "    tokens.sort()\n",
    "    return \"|\".join(tokens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "9360fc01",
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalize_tmn_n(raw):\n",
    "    return re.sub(r'.*N?(\\d|X).*', 'N\\\\1', raw.upper())\n",
    "\n",
    "def normalize_tmn_t(raw):\n",
    "    return re.sub(r'.*T?(\\d|X).*', 'T\\\\1', raw.upper())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "3570041f",
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping_GSE72970 = {\n",
    "    \"sex\": path_mapping([\"characteristics_ch1\", \"sex\"]),\n",
    "    \"age\": path_mapping([\"characteristics_ch1\", \"age\"]),\n",
    "    \"raw_drug\": path_mapping([\"characteristics_ch1\", \"regimen\"]),\n",
    "    \"response\": path_mapping([\"characteristics_ch1\", \"response status\"]),\n",
    "    'response_category': path_mapping([\"characteristics_ch1\", \"response category\"]),\n",
    "    'progression_free_survival': path_mapping([\"characteristics_ch1\", \"pfs\"]),\n",
    "    \"progression_free_survival_unit\": value(\"months\"),\n",
    "    'overall_survival': path_mapping([\"characteristics_ch1\", 'os']),\n",
    "    \"overall_survival_unit\": value(\"months\"),\n",
    "    \"sample_name\": path_mapping([\"title\", 0]),\n",
    "    \"data_processing_notes\": path_mapping([\"data_processing\", 0]),\n",
    "    \"sample_collected\": \"before_treatment\",\n",
    "    \"sample_id\": path_mapping(['geo_accession', 0]),\n",
    "    'sample_id_type': \"geo_accession\",\n",
    "    \"normalized_drug\": lambda data: normalize_drug_name(path_mapping([\"characteristics_ch1\", \"regimen\"])(data)),\n",
    "    \"source_accession\": path_mapping(['series_id', -1]),\n",
    "    \"source_type\": \"geo_series\",\n",
    "    \"platform_id\": path_mapping(['platform_id', 0]),\n",
    "    'tmn_n': lambda data: normalize_tmn_n(path_mapping([\"characteristics_ch1\", 'pn'])(data)),\n",
    "    'tmn_t': lambda data: normalize_tmn_t(path_mapping([\"characteristics_ch1\", 'pt'])(data)),\n",
    "    'tmn_m': None\n",
    "}\n",
    "\n",
    "def normalize_gse_from_GSE72970(metadata):\n",
    "    new_metadata = dict(metadata)\n",
    "    raw_characteristics = metadata.get('characteristics_ch1', [])\n",
    "    characteristics = {}\n",
    "    for item in raw_characteristics:\n",
    "        parts = item.split(':', 1)\n",
    "        key = parts[0].strip().lower()\n",
    "        value = parts[1].strip()\n",
    "        characteristics[key] = value\n",
    "    new_metadata['characteristics_ch1'] = characteristics\n",
    "    \n",
    "    return new_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a21b129d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'title': ['Primary_Tumor 01_010'],\n",
       " 'geo_accession': ['GSM1875897'],\n",
       " 'status': ['Public on Aug 17 2017'],\n",
       " 'submission_date': ['Sep 12 2015'],\n",
       " 'last_update_date': ['Aug 17 2017'],\n",
       " 'type': ['RNA'],\n",
       " 'channel_count': ['1'],\n",
       " 'source_name_ch1': ['Tissue samples taken from colorectal cancer patients'],\n",
       " 'organism_ch1': ['Homo sapiens'],\n",
       " 'taxid_ch1': ['9606'],\n",
       " 'characteristics_ch1': {'center': 'Tours',\n",
       "  'sex': 'Male',\n",
       "  'age': '53',\n",
       "  'who performance status': '0',\n",
       "  'tumor location': 'Left colon',\n",
       "  'synchronous metastase': 'Yes',\n",
       "  'pn': 'pN2',\n",
       "  'pt': 'pT3',\n",
       "  'regimen': 'FOLFOX',\n",
       "  'response category': 'PR',\n",
       "  'response status': 'R',\n",
       "  'pfs censored': '1',\n",
       "  'pfs': '5.585215569',\n",
       "  'os censored': '1',\n",
       "  'os': '5.585215569'},\n",
       " 'treatment_protocol_ch1': ['Samples were collected before chemotherapy'],\n",
       " 'molecule_ch1': ['total RNA'],\n",
       " 'extract_protocol_ch1': [\"Rneasy kit (Qiagen) for total RNA was performed according to the manufacturer's instructions.\"],\n",
       " 'label_ch1': ['biotin'],\n",
       " 'label_protocol_ch1': ['Biotinylated cRNA were prepared according to the Affymetrix protocol.'],\n",
       " 'hyb_protocol': ['Following fragmentation, cRNA were hybridized on GeneChip Human Genome U133 Plus 2.0 array.'],\n",
       " 'scan_protocol': ['GeneChips were scanned using a laser scanner.'],\n",
       " 'data_processing': ['Processing and normalization was done using the Robust Multi-array Average (RMA)'],\n",
       " 'platform_id': ['GPL570'],\n",
       " 'contact_name': ['Pierre,,Martineau'],\n",
       " 'contact_email': ['pierre.martineau@inserm.fr'],\n",
       " 'contact_phone': ['+33 4 67 61 37 43'],\n",
       " 'contact_department': ['IRCM'],\n",
       " 'contact_institute': ['Inserm'],\n",
       " 'contact_address': ['208 Rue des Apothicaires'],\n",
       " 'contact_city': ['Montpellier Cedex 5'],\n",
       " 'contact_zip/postal_code': ['34298'],\n",
       " 'contact_country': ['France'],\n",
       " 'supplementary_file': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1875nnn/GSM1875897/suppl/GSM1875897_D60726_VPA10_01_010_3mcgP.CEL.gz'],\n",
       " 'series_id': ['GSE72968', 'GSE72970'],\n",
       " 'data_row_count': ['54675']}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normalize_gse_from_GSE72970(short[0][1].metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "ac45ab01",
   "metadata": {},
   "outputs": [],
   "source": [
    "def map_sample(sample, headers, mapping, normalizer):\n",
    "    data = {}\n",
    "    metadata = normalizer(sample.metadata)\n",
    "    for header in headers:\n",
    "        mapper = mapping.get(header)\n",
    "        \n",
    "        if callable(mapper):\n",
    "            data[header] = mapper(metadata)\n",
    "        else:\n",
    "            data[header] = mapper\n",
    "    return data\n",
    "\n",
    "def make_gse_sample_records(gse_data, headers, mapping, normalizer):\n",
    "    return [map_sample(sample, headers, mapping, normalizer) for _name, sample in gse_data.gsms.items()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "0bb800ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>sample_id_type</th>\n",
       "      <th>source_accession</th>\n",
       "      <th>source_type</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>normalized_drug</th>\n",
       "      <th>raw_drug</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>data_processing_notes</th>\n",
       "      <th>...</th>\n",
       "      <th>response</th>\n",
       "      <th>response_category</th>\n",
       "      <th>progression_free_survival</th>\n",
       "      <th>progression_free_survival_unit</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>overall_survival_unit</th>\n",
       "      <th>platform_id</th>\n",
       "      <th>tmn_t</th>\n",
       "      <th>tmn_n</th>\n",
       "      <th>tmn_m</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM1875897</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Male</td>\n",
       "      <td>53</td>\n",
       "      <td>fluorouracil|leucovorin|oxaliplatin</td>\n",
       "      <td>FOLFOX</td>\n",
       "      <td>Primary_Tumor 01_010</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>5.585215569</td>\n",
       "      <td>months</td>\n",
       "      <td>5.585215569</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM1875898</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Male</td>\n",
       "      <td>57</td>\n",
       "      <td>fluorouracil|leucovorin|oxaliplatin</td>\n",
       "      <td>FOLFOX</td>\n",
       "      <td>Primary_Tumor 02_005</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>8.47638607</td>\n",
       "      <td>months</td>\n",
       "      <td>13.66735077</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM1875899</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>55</td>\n",
       "      <td>fluorouracil|leucovorin|oxaliplatin</td>\n",
       "      <td>FOLFOX</td>\n",
       "      <td>Primary_Tumor 02_006</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>NR</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.872689962</td>\n",
       "      <td>months</td>\n",
       "      <td>2.49691987</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T4</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM1875900</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>66</td>\n",
       "      <td>fluorouracil|leucovorin|oxaliplatin</td>\n",
       "      <td>FOLFOX</td>\n",
       "      <td>Primary_Tumor 02_007</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>70.01232147</td>\n",
       "      <td>months</td>\n",
       "      <td>70.01232147</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N1</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM1875901</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>56</td>\n",
       "      <td>fluorouracil|leucovorin|irinotecan</td>\n",
       "      <td>FOLFIRI</td>\n",
       "      <td>Primary_Tumor 02_008</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>7.55646801</td>\n",
       "      <td>months</td>\n",
       "      <td>19.94250488</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T4</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>GSM1876016</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>75</td>\n",
       "      <td>bevacizumab|fluorouracil|leucovorin|irinotecan</td>\n",
       "      <td>FOLFIRI+BEVACIZUMAB</td>\n",
       "      <td>Primary_Tumor 8133</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>22.30800819</td>\n",
       "      <td>months</td>\n",
       "      <td>28.35318184</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>GSM1876017</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>61</td>\n",
       "      <td>bevacizumab|fluorouracil|leucovorin|irinotecan</td>\n",
       "      <td>FOLFIRI+BEVACIZUMAB</td>\n",
       "      <td>Primary_Tumor 8135</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>NR</td>\n",
       "      <td>SD</td>\n",
       "      <td>4.43531847</td>\n",
       "      <td>months</td>\n",
       "      <td>47.86858368</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T4</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>GSM1876018</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Male</td>\n",
       "      <td>66</td>\n",
       "      <td>bevacizumab|fluorouracil|leucovorin|irinotecan</td>\n",
       "      <td>FOLFIRI+BEVACIZUMAB</td>\n",
       "      <td>Primary_Tumor 8141</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>NR</td>\n",
       "      <td>SD</td>\n",
       "      <td>7.392197132</td>\n",
       "      <td>months</td>\n",
       "      <td>7.392197132</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>GSM1876019</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Female</td>\n",
       "      <td>54</td>\n",
       "      <td>bevacizumab|fluorouracil|leucovorin|irinotecan...</td>\n",
       "      <td>FOLFIRINOX+BEVACIZUMAB</td>\n",
       "      <td>Primary_Tumor 8170</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>25.72484589</td>\n",
       "      <td>months</td>\n",
       "      <td>39.78644943</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>T3</td>\n",
       "      <td>N2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>GSM1876020</td>\n",
       "      <td>geo_accession</td>\n",
       "      <td>GSE72970</td>\n",
       "      <td>geo_series</td>\n",
       "      <td>Male</td>\n",
       "      <td>32.25462</td>\n",
       "      <td>fluorouracil|leucovorin|irinotecan|oxaliplatin</td>\n",
       "      <td>FOLFIRINOX</td>\n",
       "      <td>Primary_Tumor 8186</td>\n",
       "      <td>Processing and normalization was done using th...</td>\n",
       "      <td>...</td>\n",
       "      <td>R</td>\n",
       "      <td>PR</td>\n",
       "      <td>23.6550312</td>\n",
       "      <td>months</td>\n",
       "      <td>23.6550312</td>\n",
       "      <td>months</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>TX</td>\n",
       "      <td>NX</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>124 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample_id sample_id_type source_accession source_type     sex       age  \\\n",
       "0    GSM1875897  geo_accession         GSE72970  geo_series    Male        53   \n",
       "1    GSM1875898  geo_accession         GSE72970  geo_series    Male        57   \n",
       "2    GSM1875899  geo_accession         GSE72970  geo_series  Female        55   \n",
       "3    GSM1875900  geo_accession         GSE72970  geo_series  Female        66   \n",
       "4    GSM1875901  geo_accession         GSE72970  geo_series  Female        56   \n",
       "..          ...            ...              ...         ...     ...       ...   \n",
       "119  GSM1876016  geo_accession         GSE72970  geo_series  Female        75   \n",
       "120  GSM1876017  geo_accession         GSE72970  geo_series  Female        61   \n",
       "121  GSM1876018  geo_accession         GSE72970  geo_series    Male        66   \n",
       "122  GSM1876019  geo_accession         GSE72970  geo_series  Female        54   \n",
       "123  GSM1876020  geo_accession         GSE72970  geo_series    Male  32.25462   \n",
       "\n",
       "                                       normalized_drug  \\\n",
       "0                  fluorouracil|leucovorin|oxaliplatin   \n",
       "1                  fluorouracil|leucovorin|oxaliplatin   \n",
       "2                  fluorouracil|leucovorin|oxaliplatin   \n",
       "3                  fluorouracil|leucovorin|oxaliplatin   \n",
       "4                   fluorouracil|leucovorin|irinotecan   \n",
       "..                                                 ...   \n",
       "119     bevacizumab|fluorouracil|leucovorin|irinotecan   \n",
       "120     bevacizumab|fluorouracil|leucovorin|irinotecan   \n",
       "121     bevacizumab|fluorouracil|leucovorin|irinotecan   \n",
       "122  bevacizumab|fluorouracil|leucovorin|irinotecan...   \n",
       "123     fluorouracil|leucovorin|irinotecan|oxaliplatin   \n",
       "\n",
       "                   raw_drug           sample_name  \\\n",
       "0                    FOLFOX  Primary_Tumor 01_010   \n",
       "1                    FOLFOX  Primary_Tumor 02_005   \n",
       "2                    FOLFOX  Primary_Tumor 02_006   \n",
       "3                    FOLFOX  Primary_Tumor 02_007   \n",
       "4                   FOLFIRI  Primary_Tumor 02_008   \n",
       "..                      ...                   ...   \n",
       "119     FOLFIRI+BEVACIZUMAB    Primary_Tumor 8133   \n",
       "120     FOLFIRI+BEVACIZUMAB    Primary_Tumor 8135   \n",
       "121     FOLFIRI+BEVACIZUMAB    Primary_Tumor 8141   \n",
       "122  FOLFIRINOX+BEVACIZUMAB    Primary_Tumor 8170   \n",
       "123              FOLFIRINOX    Primary_Tumor 8186   \n",
       "\n",
       "                                 data_processing_notes  ... response  \\\n",
       "0    Processing and normalization was done using th...  ...        R   \n",
       "1    Processing and normalization was done using th...  ...        R   \n",
       "2    Processing and normalization was done using th...  ...       NR   \n",
       "3    Processing and normalization was done using th...  ...        R   \n",
       "4    Processing and normalization was done using th...  ...        R   \n",
       "..                                                 ...  ...      ...   \n",
       "119  Processing and normalization was done using th...  ...        R   \n",
       "120  Processing and normalization was done using th...  ...       NR   \n",
       "121  Processing and normalization was done using th...  ...       NR   \n",
       "122  Processing and normalization was done using th...  ...        R   \n",
       "123  Processing and normalization was done using th...  ...        R   \n",
       "\n",
       "    response_category progression_free_survival  \\\n",
       "0                  PR               5.585215569   \n",
       "1                  PR                8.47638607   \n",
       "2                  PD               1.872689962   \n",
       "3                  PR               70.01232147   \n",
       "4                  PR                7.55646801   \n",
       "..                ...                       ...   \n",
       "119                PR               22.30800819   \n",
       "120                SD                4.43531847   \n",
       "121                SD               7.392197132   \n",
       "122                PR               25.72484589   \n",
       "123                PR                23.6550312   \n",
       "\n",
       "    progression_free_survival_unit overall_survival overall_survival_unit  \\\n",
       "0                           months      5.585215569                months   \n",
       "1                           months      13.66735077                months   \n",
       "2                           months       2.49691987                months   \n",
       "3                           months      70.01232147                months   \n",
       "4                           months      19.94250488                months   \n",
       "..                             ...              ...                   ...   \n",
       "119                         months      28.35318184                months   \n",
       "120                         months      47.86858368                months   \n",
       "121                         months      7.392197132                months   \n",
       "122                         months      39.78644943                months   \n",
       "123                         months       23.6550312                months   \n",
       "\n",
       "    platform_id tmn_t tmn_n tmn_m  \n",
       "0        GPL570    T3    N2  None  \n",
       "1        GPL570    T3    N2  None  \n",
       "2        GPL570    T4    N2  None  \n",
       "3        GPL570    T3    N1  None  \n",
       "4        GPL570    T4    N2  None  \n",
       "..          ...   ...   ...   ...  \n",
       "119      GPL570    T3    N2  None  \n",
       "120      GPL570    T4    N2  None  \n",
       "121      GPL570    T3    N2  None  \n",
       "122      GPL570    T3    N2  None  \n",
       "123      GPL570    TX    NX  None  \n",
       "\n",
       "[124 rows x 21 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GSE72970_meta = pd.DataFrame.from_records(make_gse_sample_records(gse, headers, mapping_GSE72970, normalize_gse_from_GSE72970))\n",
    "GSE72970_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "21e642e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "GSE72970_expression = to_expression_matrix(gse, probe_mappings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "e534baa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "GSE72970_expression.to_csv('./GSE72970_expression_matrix_raw.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "6d6cddbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "GSE72970_meta.to_csv('./GSE72970_sample_metadata_raw.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
